src/common/debug.c

```diff
@@ -68,6 +78,34 @@ static void __oscap_debuglog_close(void)
         fclose(__debuglog_fp);
 }
 
+int oscap_verbosity_level_from_cstr(const char *level_name)
+{
+	return oscap_string_to_enum(OSCAP_VERBOSITY_LEVELS, level_name);
+}
+
+bool oscap_set_verbose(const char* verbosity_level, const char *filename, bool is_probe)
+{
+	if (verbosity_level == NULL || filename == NULL) {
+		return false;
+	}
+	if (is_probe) {
+		__debuglog_fp = fopen(filename, "a");
+	} else {
+		setenv("OSCAP_PROBE_VERBOSITY_LEVEL", verbosity_level, 1);
+		setenv("OSCAP_PROBE_VERBOSE_LOG_FILE", filename, 1);
+		__debuglog_fp = fopen(filename, "w");
+	}
+	if (__debuglog_fp == NULL) {
+		oscap_seterr(OSCAP_EFAMILY_OSCAP, "Failed to open file %s.", filename);
+		return false;
+	}
+	setbuf(__debuglog_fp, NULL);
+	__debuglog_level = oscap_verbosity_level_from_cstr(verbosity_level);
```

Comment:
I would prefer to move this line upper. That way you can check what user entered. If they enters 'BLA' you may want to stop proceeding (i.e. not open any file).